impeller/entity/contents/color_source_contents.h

```diff
@@ -100,6 +101,104 @@ class ColorSourceContents : public Contents {
   // |Contents|
   void SetInheritedOpacity(Scalar opacity) override;
 
+ protected:
+  using BindFragmentCallback = std::function<bool(RenderPass& pass)>;
+  using PipelineBuilderMethod = std::shared_ptr<Pipeline<PipelineDescriptor>> (
+      impeller::ContentContext::*)(ContentContextOptions) const;
+  using PipelineBuilderCallback =
+      std::function<std::shared_ptr<Pipeline<PipelineDescriptor>>(
+          ContentContextOptions)>;
+
+  template <typename VertexShaderT>
+  bool DrawGeometry(GeometryResult geometry_result,
+                    const ContentContext& renderer,
+                    const Entity& entity,
+                    RenderPass& pass,
+                    const PipelineBuilderCallback& pipeline_callback,
+                    typename VertexShaderT::FrameInfo frame_info,
+                    const BindFragmentCallback& bind_fragment_callback) const {
+    auto options = OptionsFromPassAndEntity(pass, entity);
+
+    // If overdraw prevention is enabled (like when drawing stroke paths), we
+    // increment the stencil buffer as we draw, preventing overlapping fragments
+    // from drawing. Afterwards, we need to append another draw call to clean up
+    // the stencil buffer (happens below in this method).
+    if (geometry_result.prevent_overdraw) {
+      options.stencil_mode =
+          ContentContextOptions::StencilMode::kLegacyClipIncrement;
+    }
+    options.primitive_type = geometry_result.type;
+    pass.SetVertexBuffer(std::move(geometry_result.vertex_buffer));
+    pass.SetStencilReference(entity.GetClipDepth());
+
+    // Take the pre-populated vertex shader uniform struct and set managed
+    // values.
+    frame_info.depth = entity.GetShaderClipDepth();
+    frame_info.mvp = geometry_result.transform;
+
+    VertexShaderT::BindFrameInfo(
+        pass, renderer.GetTransientsBuffer().EmplaceUniform(frame_info));
+
+    std::shared_ptr<Pipeline<PipelineDescriptor>> pipeline =
+        pipeline_callback(options);
+    pass.SetPipeline(pipeline);
```

Comment:
Either move into SetPipeline or inline the call to avoid the additional shared_ptr release.

Reply:
Oopsies, done.